Sources/Mimus/Storage.swift

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright (©) 2019 AirHelp. All rights reserved.
+//
+
+import Foundation
+
+/// Structure used to hold recorded invocations
+struct RecordedCall {
+    let identifier: String
+    let arguments: [Any?]?
+}
+
+struct RecordedReturnEntry {
+    let returnValue: Any?
+    let recordedCall: RecordedCall
+    let thrownError: Error?
+}
+
+/// Class responsible for storing recorded calls.
+public class Storage {
+    var recordedCalls: [RecordedCall]
+    var recordedReturnValueEntries: [RecordedReturnEntry]
+
+    public init() {
+        recordedCalls = []
+        recordedReturnValueEntries = []
+    }
+
+    func record(call: RecordedCall) {
+        recordedCalls.append(call)
+    }
+
+    func record(entry: RecordedReturnEntry) {
+        recordedReturnValueEntries.append(entry)
+    }
+
+    /// Resets all recorded calls. Use to return given mock to initial state.
+    public func reset() {
+        recordedCalls = []
+        recordedReturnValueEntries = []
+    }
+}
+
+extension Storage {
+    func recordedReturnEntry(matchingIdentifier identifier: String, arguments: Arguments) -> RecordedReturnEntry? {
```

Comment:
Some tests would be great here as well.
Especially as a context: explaining how to use this method.

Reply:
Same here, this will be tested by tests for `Mock`.

Reply:
👍